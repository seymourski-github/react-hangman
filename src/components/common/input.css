.input {
  height: 2em;
}
