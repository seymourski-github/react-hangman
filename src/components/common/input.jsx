import React from 'react';
import PropTypes from 'prop-types';
import './input.css';

const onBlur = handleInput => e => handleInput(e);
const onChange = handleInput => e => handleInput(e);

const mapProps = ({ autoFocus, className, disabled, id, name, type, value }) => ({
  id: id || name,
  autoFocus, className, disabled, name, type, value
});

const mapStyle =({ flexItem, style }) => ({
  ...style,
  flex: flexItem
});

export const Input = props => {
  const { handleBlur, handleChange } = props;
  return (
    <input {...mapProps(props)}
      style={mapStyle(props)}
      onBlur={handleBlur && onBlur(handleBlur)}
      onChange={handleChange && onChange(handleChange)}
    />
  );
};

Input.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  flexItem: PropTypes.string,
  style: PropTypes.object,
  type: PropTypes.string,
  value: PropTypes.string,
  handleBlur: PropTypes.func,
  handleChange: PropTypes.func,
};

Input.defaultProps = {
  handleBlur: e => false,
  handleChange: e => false,
  className: 'input',
  disabled: false,
  id: '',
  type: 'text',
  style: null,
  value: ''
}
