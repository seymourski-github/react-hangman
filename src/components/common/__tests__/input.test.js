import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Input } from '../input';

const props = {
  name: 'myTextInput',
  value: 'My text input value'
};

describe('<Input /> Component', function() {
  const component = renderer.create(<Input {...props} />);
  const tree = component.toJSON();

  it("renders and matches our snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
