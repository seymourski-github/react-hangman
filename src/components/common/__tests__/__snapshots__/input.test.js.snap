// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Component renders and matches our snapshot 1`] = `
<input
  autoFocus={undefined}
  className="input"
  disabled={false}
  id="myTextInput"
  name="myTextInput"
  onBlur={[Function]}
  onChange={[Function]}
  style={
    Object {
      "flex": undefined,
    }
  }
  type="text"
  value="My text input value"
/>
`;
